### PR TITLE
docs(segmented-button): Add README.md

### PR DESCRIPTION
**Pull request by gschrag**
Include README instructions and documentation

### CODE REVIEW

File: packages/mdc-segmented-button/README.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+```
+
+## Basic Usage
+
+### HTML Structure
+
+```html
+<div class="mdc-segmented-button" role="group">
+  <button class="mdc-segmented-button__segment" aria-pressed="false">
+      <i class="material-icons mdc-segmented-button__icon">favorite</i>
```
Comment (kseamon) [inline packages/mdc-segmented-button/README.md:20]: nit: use consistent indentation (2 spaces probably)

(here and below)

**Comment by abhiomkar** [comment]
Adding @joyzhong as reviewer for Dev Docs expertise.

File: packages/mdc-segmented-button/README.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+<button class="mdc-segmented-button__segment mdc-segmented-button__segment--selected" aria-pressed="true">
+  <div class="mdc-segmented-button__label">Sample Text</div>
+</button>
+```
+
+## Additional Information
+
+### Touch accessibility
+
+`//TODO: update this section if incorrect`
```
Comment (abhiomkar) [inline packages/mdc-segmented-button/README.md:150]: Remove TODO comment.

File: packages/mdc-segmented-button/README.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+</div>
+```
+
+### Styles
+
+```scss
+@use "@material/segmented-button/segmented-button";
+@use "@material/segmented-button/segment";
+
+@include segmented-button.core-styles();
```
Comment (abhiomkar) [inline packages/mdc-segmented-button/README.md:38]: Let's update this section to reflect the recent change about MDC ripple common styles.

File: packages/mdc-segmented-button/README.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+@include segment.core-styles();
```
Comment (abhiomkar) [inline packages/mdc-segmented-button/README.md:39]: Replace with

```scss
@use "@material/ripple/common";
@use "@material/segmented-button/styles";
```

**Pull request merged**